modules/crop-workflowoperation/src/main/java/org/opencastproject/workflow/handler/crop/CropWorkflowOperationHandler.java

```diff
@@ -0,0 +1,199 @@
+
+/**
+ * The workflow definition will run recordings to crop them from 4:3 to 16:9.
+ */
+public class CropWorkflowOperationHandler extends AbstractWorkflowOperationHandler {
+
+  /** The logging facility */
+  private static final Logger logger = LoggerFactory.getLogger(CropWorkflowOperationHandler.class);
+
+  /** Name of the configuration key that specifies the flavor of the track to analyze */
+  private static final String PROP_SOURCE_FLAVOR = "source-flavor";
+
+  private static final String PROP_TARGET_FLAVOR = "target-flavor";
+
+  /** Name of the configuration key that specifies the flavor of the track to analyze */
+  private static final String PROP_TARGET_TAGS = "target-tags";
+
+  /** Id builder used to create ids for cropped tracks */
+  private final IdBuilder idBuilder = IdBuilderFactory.newInstance().newIdBuilder();
+
+  /** The composer service */
+  private CropService cropService = null;
+
+  /** The local workspace */
+  private Workspace workspace = null;
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance, JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobContext jobContext)
+          throws WorkflowOperationException {
+
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
+    MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+
+    logger.info("Start cropping workflow operation for mediapackage {}", mediaPackage.getIdentifier().compact());
+
+    List<String> targetTags = asList(operation.getConfiguration(PROP_TARGET_TAGS));
+
+    MediaPackageElementFlavor targetFlavor = null;
+    String targetFlavourText = StringUtils.trimToNull(operation.getConfiguration(PROP_TARGET_FLAVOR));
+    if (targetFlavourText != null) {
+      try {
+        targetFlavor = MediaPackageElementFlavor.parseFlavor(targetFlavourText);
+      } catch (IllegalArgumentException e) {
+        throw new WorkflowOperationException("Target flavor is malformed");
+      }
+    }
+
+    String trackFlavor = StringUtils.trimToNull(operation.getConfiguration(PROP_SOURCE_FLAVOR));
+    if (trackFlavor == null) {
+      throw new WorkflowOperationException(String.format("Required property %s not set", PROP_SOURCE_FLAVOR));
+    }
+
+    List<Track> candidates = new ArrayList<>();
+    candidates.addAll(Arrays.asList(mediaPackage.getTracks(MediaPackageElementFlavor.parseFlavor(trackFlavor))));
+    candidates.removeIf(t -> !t.hasVideo());
+
+    if (candidates.size() == 0) {
+      logger.info("No matching tracks available for cropping in workflow {}", workflowInstance);
+      return createResult(WorkflowOperationResult.Action.CONTINUE);
+    } else if (candidates.size() > 1) {
+      logger.info("Found more than one track to crop");
+    }
+
+    // start cropping all candidates in parallel
+    //List<Job> jobs = new ArrayList<>();
```

Comment:
I think this needs to be deleted. ;)

Reply:
Definitely Christian ;-). 
Thanks a lot for the first review.